### PR TITLE
feat(eslint-plugin): add rule `prefer-ts-expect-error`

### PR DESCRIPTION
**Pull request by G-Rath**
I think this should be included in the recommended, as I can't think of a case where `// @ts-expect-error` can't replace `// @ts-ignore`, but that might get tricky since TS3.9 & ESLint 7 are a few months apart 😬

### CODE REVIEW

File: packages/typescript-estree/src/ts-estree/ts-estree.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export interface LineComment extends BaseToken {
 export type Comment = BlockComment | LineComment;
 export type Token =
   | BooleanToken
+  | Comment
```
Comment (G-Rath) [inline packages/typescript-estree/src/ts-estree/ts-estree.ts]: Seems this was left out of the union - might have been on purpose but can't find any reference to it.

[ESLint passes these nodes to `fixer` methods](https://github.com/eslint/eslint/blob/master/lib/rules/multiline-comment-style.js#L393), so I think it's fine.

Happy to break this out into another PR if you'd like to land it for the next release without having to review everything else :)

**Comment by bradzacher** [comment]
There's been a few linting changes since this was last merged - once those are resolved, this is good to merge

**Comment by G-Rath** [comment]
> There's been a few linting changes

I noticed a few new internal rules - I'm a big fan of the ones around applying prettier to all tests, and the aligning of multiline tests - might have to pinch those from you 😉

**Comment by bradzacher** [comment]
I should probably put that rule somewhere more accessible - I'm sure there's a few people that wouldn't mind it. It took me a few hours to write it!
